Source/Core/Common/BitFieldView.h

```diff
@@ -0,0 +1,788 @@
+  if constexpr (std::is_signed_v<field_t>)
+    return static_cast<field_t>(static_cast<shost_t>(host << lshift) >> rshift);
+  else
+    return static_cast<field_t>(static_cast<uhost_t>(host << lshift) >> rshift);
+}
+
+template <typename field_t, std::size_t start, std::size_t width, typename host_t>
+constexpr void SetBitFieldFixed(host_t& host, const field_t val)
+{
+  AssertHostTypeIsSane<host_t>();
+  AssertFieldTypeIsSane<field_t>();
+  AssertBitFieldFixedIsSane<field_t, start, width, host_t>();
+
+  using uhost_t = std::make_unsigned_t<host_t>;
+  constexpr std::size_t rshift = 8 * sizeof(host_t) - width;
+  constexpr uhost_t mask = std::numeric_limits<uhost_t>::max() >> rshift << start;
+
+  host = (host & ~mask) | ((static_cast<uhost_t>(val) << start) & mask);
+}
+
+template <typename field_t, typename host_t>
+constexpr void SetBitField(const std::size_t start, const std::size_t width, host_t& host,
+                           const field_t val)
+{
+  AssertHostTypeIsSane<host_t>();
+  AssertFieldTypeIsSane<field_t>();
+  AssertBitFieldIsSane<field_t, host_t>(start, width);
+
+  using uhost_t = std::make_unsigned_t<host_t>;
+  const std::size_t rshift = 8 * sizeof(host_t) - width;
+  const uhost_t mask = std::numeric_limits<uhost_t>::max() >> rshift << start;
+
+  host = (host & ~mask) | ((static_cast<uhost_t>(val) << start) & mask);
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Float host type specialization
+
+template <typename field_t, std::size_t start, std::size_t width>
+constexpr field_t GetBitFieldFixed(const float& host)
+{
+  AssertFieldTypeIsSane<field_t>();
+  AssertBitFieldFixedIsSane<field_t, start, width, float>();
+
+  constexpr std::size_t rshift = 8 * sizeof(float) - width;
+  constexpr std::size_t lshift = rshift - start;
+
+  std::uint32_t hostbits = Common::BitCast<std::uint32_t>(host);
+  if constexpr (std::is_signed_v<field_t>)
+    return static_cast<field_t>(static_cast<std::int32_t>(hostbits << lshift) >> rshift);
+  else
+    return static_cast<field_t>(static_cast<std::uint32_t>(hostbits << lshift) >> rshift);
+}
+
+template <typename field_t>
+constexpr field_t GetBitField(const std::size_t start, const std::size_t width, const float& host)
+{
+  AssertFieldTypeIsSane<field_t>();
+  AssertBitFieldIsSane<field_t, float>(start, width);
+
+  const std::size_t rshift = 8 * sizeof(float) - width;
+  const std::size_t lshift = rshift - start;
+
+  std::uint32_t hostbits = Common::BitCast<std::uint32_t>(host);
+  if constexpr (std::is_signed_v<field_t>)
+    return static_cast<field_t>(static_cast<std::int32_t>(hostbits << lshift) >> rshift);
+  else
+    return static_cast<field_t>(static_cast<std::uint32_t>(hostbits << lshift) >> rshift);
+}
+
+template <typename field_t, std::size_t start, std::size_t width>
+constexpr void SetBitFieldFixed(float& host, const field_t val)
```

Comment:
If `SetBitFieldFixed` instead returned it's new value, this can be simplified like so:
```cpp
template <typename field_t, std::size_t start, std::size_t width>
constexpr float SetBitFieldFixed(float host, const field_t val)
{
   return Common::BitCast<float>(SetBitFieldFixed<field_t, start, width>(Common::BitCast<uint32_t>(host), val));
}
```
This would be slightly more DRY, and I personally find this preferrable.

Reply:
I found a DRY solution which is less of an upheaval.